src/deadline/job_attachments/api/manifest.py

```diff
@@ -128,3 +173,78 @@ def _manifest_snapshot(
+    force_rehash=False,
+    logger: ClickLogger = ClickLogger(False),
+) -> ManifestDiff:
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to diff a manifest root with a previously snapshotted manifest.
+    :param manifest: Manifest file path to compare against.
+    :param root: Root directory to generate the manifest fileset.
+    :param include: Include glob to look for files to add to the manifest.
+    :param exclude: Exclude glob to exclude files from the manifest.
+    :param include_exclude_config: Config JSON or file containeing input and exclude config.
+    :param logger: Click Logger instance to print to CLI as test or JSON.
+    :returns: ManifestDiff object containing all new changed, deleted files.
+    """
+
+    # Find all files matching our regex
+    input_files = _glob_files(
+        root=root, include=include, exclude=exclude, include_exclude_config=include_exclude_config
+    )
+    input_paths = [Path(p) for p in input_files]
+
+    # Placeholder Asset Manager
+    asset_manager = S3AssetManager()
+
+    # parse the given manifest to compare against.
+    local_manifest_object: BaseAssetManifest
+    with open(manifest) as input_file:
+        manifest_data_str = input_file.read()
+        local_manifest_object = decode_manifest(manifest_data_str)
+
+    output: ManifestDiff = ManifestDiff()
+
+    if force_rehash:
+        # hash and create manifest of local directory
+        cache_config = config_file.get_cache_directory()
+        with HashCache(cache_config) as hash_cache:
+            directory_manifest_object = asset_manager._create_manifest_file(
+                input_paths=input_paths, root_path=root, hash_cache=hash_cache
+            )
+
+        # Hash based compare manifests.
+        differences: List[Tuple[FileStatus, BaseManifestPath]] = compare_manifest(
+            reference_manifest=local_manifest_object, compare_manifest=directory_manifest_object
+        )
+        # Map to output datastructure.
+        for item in differences:
+            if item[0] == FileStatus.MODIFIED:
+                output.modified.append(item[1].path)
+            elif item[0] == FileStatus.NEW:
+                output.new.append(item[1].path)
+            elif item[0] == FileStatus.DELETED:
+                output.deleted.append(item[1].path)
+
+    else:
+        # File based comparisons.
+        fast_diff: List[Tuple[str, FileStatus]] = _fast_file_list_to_manifest_diff(
+            root=root, current_files=input_files, diff_manifest=local_manifest_object, logger=logger
+        )
+        for fast_diff_item in fast_diff:
+            if fast_diff_item[1] == FileStatus.MODIFIED:
+                output.modified.append(fast_diff_item[0])
+            elif fast_diff_item[1] == FileStatus.NEW:
+                output.new.append(fast_diff_item[0])
+            elif fast_diff_item[1] == FileStatus.DELETED:
+                output.deleted.append(fast_diff_item[0])
```

Comment:
The comparison for both cases look very similar, maybe we could refactor or use helper?

Reply:
Yes and no, the loop is the same, but the data structure it processes is the different. So I dont' see much refactoring options here.

Reply:
Maybe something like?
 
```
def process_output(status, path, output):
    if status == FileStatus.MODIFIED:
        output.modified.append(path)
    elif status == FileStatus.NEW:
        output.new.append(path)
    elif status == FileStatus.DELETED:
        output.deleted.append(path)
    else:
        raise InvalidStatusError()
```

Reply:
Sure, I can do that cleanup but I wanted to avoid nested functions :(